src/Extension/Challenge.cs

```diff
@@ -48,34 +51,13 @@ public async Task InvokeOnCodeSubmittedHandler()
             await OnCodeSubmittedHandler(_context);
         }
 
-        public Evaluation EvaluateChallengeEvaluationByDefault(RuleContext result)
+        private void EvaluateChallengeEvaluationByDefault()
         {
-            // todo: result unused
-            // prob remove this arg because 
-            // we'll use challenge info in this object to construct rulecontext
-            // to pass them into rule.Evaluate()
-
-            // todo: two pathways: teacher sets by using challengeContext.Fail, etc
-            // or default behavior, which is this function
-            var evaluation = new Evaluation();
+            // todo: discussion, is this right default behavior?
 
-            var listOfRulePassOrFailOutcomes = new List<bool>();
-            foreach (var rule in _rules)
-            {
-                var ruleContext = new RuleContext();
-                rule.Evaluate(ruleContext);
-                listOfRulePassOrFailOutcomes.Add(ruleContext.Passed);
-            }
-
-            if (listOfRulePassOrFailOutcomes.Contains(false))
-            {
-                evaluation.SetOutcome(Outcome.Failure);
-            }
-            else
-            {
-                evaluation.SetOutcome(Outcome.Success);
-            }
-            return evaluation;
+            var didAllRulesPass = CurrentEvaluation.RuleEvaluations
+                .Select(e => e.Outcome).All(o => o == Outcome.Success);
+            CurrentEvaluation.SetOutcome(didAllRulesPass ? Outcome.Success : Outcome.Failure);
```

Comment:
this behavior should be worth discussing. I've written it like this just as a continuation of the previous behavior. I've added this as an open question in the sprint plan